sbol2/test/test_object.py

```diff
@@ -173,3 +173,49 @@ def test_this(self):
             self.assertEqual(md.this, md)
         # Verify that a warning was issued
         self.assertEqual(len(warns), 1)
+
+    def test_compare_properties(self):
+        # Test that two objects are equal even if a property is in a
+        # different order
+        expected1 = [sbol2.SBO_INHIBITION, sbol2.SBO_CONTROL]
+        expected2 = [sbol2.SBO_CONTROL, sbol2.SBO_INHIBITION]
+        md1 = sbol2.ModuleDefinition('md1')
+        md1.roles = expected1
+        md2 = sbol2.ModuleDefinition('md1')
+        md2.roles = expected2
+        self.assertTrue(md1.compare(md2))
+
+    def test_compare_owned_objects(self):
+        # Test that two objects are equal even if owned objects are in
+        # a different order
+        expected1 = [sbol2.Module('m1'), sbol2.Module('m2')]
+        expected2 = [sbol2.Module('m2'), sbol2.Module('m1')]
+        md1 = sbol2.ModuleDefinition('md1')
+        md1.modules = expected1
+        md2 = sbol2.ModuleDefinition('md1')
+        md2.modules = expected2
+        self.assertTrue(md1.compare(md2))
+
+    def test_compare_recursive(self):
```

Comment:
I'd be more reassured that the comparison is truly recursive if we were comparing object trees with 3-levels of depth rather than 2.  (For example, it would be really easy for an oaf like me to write a non-recursive comparison operator that could pass this test).  Don't know if that's really worth the extra effort though.